extensions/spring-web/deployment/src/main/java/io/quarkus/spring/web/deployment/ControllerAdviceAbstractExceptionMapperGenerator.java

```diff
@@ -24,23 +33,77 @@
     private static final DotName RESPONSE_ENTITY = DotName.createSimple("org.springframework.http.ResponseEntity");
 
     private final MethodInfo controllerAdviceMethod;
-    private final IndexView index;
+    private final TypesUtil typesUtil;
     private final Type returnType;
-    private final Type parameterType;
+    private final List<Type> parameterTypes;
     private final String declaringClassName;
 
+    private final Map<Type, FieldDescriptor> parameterTypeToField = new HashMap<>();
+
     ControllerAdviceAbstractExceptionMapperGenerator(MethodInfo controllerAdviceMethod, DotName exceptionDotName,
-            ClassOutput classOutput, IndexView index) {
+            ClassOutput classOutput, TypesUtil typesUtil) {
         super(exceptionDotName, classOutput);
         this.controllerAdviceMethod = controllerAdviceMethod;
-        this.index = index;
+        this.typesUtil = typesUtil;
 
         this.returnType = controllerAdviceMethod.returnType();
-        this.parameterType = controllerAdviceMethod.parameters().size() == 0 ? null
-                : controllerAdviceMethod.parameters().get(0);
+        this.parameterTypes = controllerAdviceMethod.parameters();
         this.declaringClassName = controllerAdviceMethod.declaringClass().name().toString();
     }
 
+    /**
+     * We need to go through each parameter of the method of the ControllerAdvice
+     * and make sure it's supported
+     * The javax.ws.rs.ext.ExceptionMapper only has one parameter, the exception, however
+     * other parameters can be obtained using @Context and therefore injected into the target method
+     */
+    @Override
+    protected void preGenerateMethodBody(ClassCreator cc) {
+        int notAllowedParameterIndex = -1;
+        for (int i = 0; i < parameterTypes.size(); i++) {
+            Type parameterType = parameterTypes.get(i);
+            DotName parameterTypeDotName = parameterType.name();
+            if (typesUtil.isAssignable(Exception.class, parameterTypeDotName)) {
+                // do nothing since this will be handled during in generateMethodBody
+            } else if (typesUtil.isAssignable(HttpServletRequest.class, parameterTypeDotName)) {
+                if (parameterTypeToField.containsKey(parameterType)) {
+                    throw new IllegalArgumentException("Parameter type " + parameterTypes.get(notAllowedParameterIndex).name()
+                            + " is being used multiple times in method" + controllerAdviceMethod.name() + " of class"
+                            + controllerAdviceMethod.declaringClass().name());
+                }
+
+                // we need to generate a field that injects the HttpServletRequest into the class
+                FieldCreator httpRequestFieldCreator = cc.getFieldCreator("httpServletRequest", HttpServletRequest.class)
+                        .setModifiers(Modifier.PRIVATE);
+                httpRequestFieldCreator.addAnnotation(Context.class);
```

Comment:
It might be a stupid question but is the scope of the `ExceptionMapper` restricted enough so that we can inject context values in it? I have a vague recollection of discussion about that but I might be mistaken.

Reply:
I am sure it's a very valid question butI am not sure I follow. What could the potential problem be?

Reply:
If the exception mapper is scoped singleton or application, I'm not sure it's safe to inject context values in its fields. Or maybe RESTEasy takes care of that.

Reply:
I didn't encounter any problems TBH and looking around it seemed like people were using this solution when writing their own ExceptionMappers

Reply:
I think we should be OK for the time being. We can always go deeper if users report errors. WDYT  @gsmet ?